### PR TITLE
Quick Sort 3-way in Java (#2170)

### LINKED ISSUE #2170: Quick Sort 3 way
- [ ] C - @Moddy2024
- [x] C++
- [ ] C# - @Harikrishnan6336
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [ ] Java - @IPSITA2000
- [ ] JS - @atrip10
- [ ] Kotlin
- [ ] Php - @BellatrixLestrangee
- [x] Python
- [ ] Readme - @SomyaGandhi
- [x] Ruby - @HariniJeyaraman
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by IPSITA2000**
Fixes #2170

#### Checklist:
- [ *] 4 space indentation.
- [ *]Coding conventions are followed.
- [ *] Input is taken dynamically.
- [ *] Sample Input / Output is added at the end of file.
- [ *] Logic Documentation (Comments).
- [ *]File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- Java
- 

#### Files Added:
- Quicksort_3_way.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by IPSITA2000** [comment]
@jainaman224 Please review my PR.

File: Quicksort_3_way/Quicksort_3_way.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import java.util.Scanner;
+import java.lang.Math;
+
+public class practice1{
+
+    public static void main (String[] args){
+
+        //        int left[] = new int [];
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Remove these extra spaces after `//`. There should be only one space.

File: Quicksort_3_way/Quicksort_3_way.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import java.util.Scanner;
+import java.lang.Math;
+
+public class practice1{
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Add a space before `{`.

File: Quicksort_3_way/Quicksort_3_way.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import java.util.Scanner;
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Add documentation at the top.

File: Quicksort_3_way/Quicksort_3_way.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+        //        int left[] = new int [];
+
+        Scanner input = new Scanner(System.in);
+
+        System.out.print("Enter numbers separated by spaces: ");
+        String elements = input.nextLine();
+
+        //break up numbers and put into array
+        String arr[]= elements.split(" ");
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Add space before `=`

File: Quicksort_3_way/Quicksort_3_way.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        int intArr[] = new int[arr.length];
+        for (int i = 0; i < intArr.length; ++i) {
+            intArr[i] = Integer.parseInt(arr[i]);
+            //            System.out.printf("%d\n", intArr[i]);
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Remove extra spaces after `//`

File: Quicksort_3_way/Quicksort_3_way.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        }
+        
+        int size = intArr.length;
+             
+        //3-way partition
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Add one space after `//`

File: Quicksort_3_way/Quicksort_3_way.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        int i = 0;
+        int j = 0; 
+        int n = size - 1;
+        int temp;
+        System.out.print("Original Array: ");
+        for ( int k = 0; k < size; k++) 
+            System.out.printf("%d ", intArr[k]);
+        System.out.printf("\nThe pivot is: %d\n", pivot);
+
+
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Remove this extra new-line.

File: Quicksort_3_way/Quicksort_3_way.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        while(j <= n){
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Add space between `)` and `{`

File: Quicksort_3_way/Quicksort_3_way.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+            if (intArr[j] < pivot){
+                temp = intArr[i];
+                intArr[i] = intArr[j];
+                intArr[j] = temp;
+                i+=1;
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Add space around operators.

File: Quicksort_3_way/Quicksort_3_way.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+                j+=1;
+
+        }
+
+        System.out.print("Sorted Array: ");
+        for ( int k = 0; k < size; k++)
+            System.out.printf("%d ", intArr[k]);
+        System.out.printf("\n");
+    }
+}
```
Comment (SakshiShreya) [inline Quicksort_3_way/Quicksort_3_way.java]: Add a sample input and output at the end of the file.

Comment (IPSITA2000) [reply Quicksort_3_way/Quicksort_3_way.java]: Ok ma'am , i will make the corrections at the earliest.

**Comment by MastersAbh** [comment]
Hi @IPSITA2000 I dont see any update for 21 days, hence I am closing this PR. Please reach out if you still want to work on this.

**Pull request closed without merging**